[Backend] - Bug creating classes

When i'm trying to make a second class for a certain teacher with a certain id i get:
```json
{"code":"INTERNAL_ERROR","message":"Unexpected server error"}
```
When looking at the database log i see:
```bash
backend-1   | query failed: INSERT INTO "teacher_of_class_type_orm"("id", "since", "teacher_id", "class_id") VALUES (DEFAULT, DEFAULT, $1, $2) RETURNING "id", "since" -- PARAMETERS: ["219c1e2f-488a-4f94-a9d8-38b7c9bede1f","c5a54bb4-a43c-4db9-8e36-895d07ae2eeb"]
backend-1   | error: error: duplicate key value violates unique constraint "REL_21b1af017293b9fadc8cedc6a8"
backend-1   | query: ROLLBACK
backend-1   | [2025-03-31T19:56:59.654] [ERROR] error - Server error (23505): duplicate key value violates unique constraint "REL_21b1af017293b9fadc8cedc6a8"
```

So it seems that it isn't possible that a teacher has more than 1 class which shouldn't be.


- [ ] Fix bug: est. time: 1h

--- a/backend/src/infrastructure/database/data/data_models/teacherOfClassTypeorm.ts
+++ b/backend/src/infrastructure/database/data/data_models/teacherOfClassTypeorm.ts
@@ -1,4 +1,4 @@
-import { Entity, PrimaryGeneratedColumn, OneToOne, JoinColumn, CreateDateColumn } from "typeorm";
+import { Entity, PrimaryGeneratedColumn, ManyToOne, JoinColumn, CreateDateColumn } from "typeorm";
 import { ClassTypeORM } from "./classTypeorm";
 import { TeacherTypeORM } from "./teacherTypeorm";
 
@@ -7,11 +7,11 @@ export class TeacherOfClassTypeORM {
     @PrimaryGeneratedColumn("uuid")
     id!: string;
 
-    @OneToOne(() => TeacherTypeORM)
+    @ManyToOne(() => TeacherTypeORM)
     @JoinColumn({ name: "teacher_id" })
     teacher!: TeacherTypeORM;
 
-    @OneToOne(() => ClassTypeORM)
+    @ManyToOne(() => ClassTypeORM)
     @JoinColumn({ name: "class_id" })
     class!: ClassTypeORM;
 

